js/web/lib/wasm/jsep/webgpu/ops/argminmax.ts

```diff
@@ -0,0 +1,156 @@
+  keepDims: boolean;
+  axes: number;
+  selectLastIndex: number;
+}
+
+type ArgMinMaxOp = (inputs: readonly TensorView[], axes: number[]) => string[];
+
+const createReduceProgramInfo =
+    (metadata: ProgramMetadata, inputs: readonly TensorView[], attributes: ArgMinMaxAttributes,
+     argMinMaxOp: ArgMinMaxOp): ProgramInfo => {
+      const outputShape: number[] = [];
+      const inputShape = inputs[0].dims;
+
+      const idxCopy: string[] = [];  // copy output indexes to input indexes
+
+      const axes = ShapeUtil.normalizeAxes([attributes.axes], inputs[0].dims.length);
+      const outputDimsLength = inputs[0].dims.length - (attributes.keepDims ? 0 : axes.length);
+      const ops = argMinMaxOp(inputs, axes);
+      const inputIndicesHelper = createIndicesHelper('input', inputShape);
+      const initInputIdx = (ops[1] === '') ? '' : `let inputIdx = ${inputIndicesHelper.i2oExpression('inputIndices')};`;
+      let reduceOps = `
+          let inputIdx = ${inputIndicesHelper.i2oExpression('inputIndices')};
+          ${ops[2]};`;
+      for (let k = 0; k < inputs[0].dims.length; k++) {
+        // if this axis is reduced
+        if (axes.indexOf(k) >= 0) {
+          if (attributes.keepDims) {
+            outputShape.push(1);
+          }
+          // loop over the d-th axis
+          reduceOps = `for(var j${k}: u32 = 0; j${k} < ${inputs[0].dims[k]}; j${k}++) {
+                            let lastIndex = j${k};
+                            inputIndices[${k}] = lastIndex;
+                            ${reduceOps}
+                          }`;
+        } else {
+          if (outputDimsLength > 1) {
+            idxCopy.push(`inputIndices[${k}] = outputIndices[${outputShape.length}];`);
+          } else {
+            idxCopy.push(`inputIndices[${k}] = outputIndices;`);
+          }
+          outputShape.push(inputs[0].dims[k]);
+        }
+      }
+
+      const outputIndicesHelper = createIndicesHelper('output', outputShape);
+      const outputSize = ShapeUtil.size(outputShape);
+      const dataType = 'f32';
+
+      const getShaderSource = (shaderHelper: ShaderHelper) => `
+          @group(0) @binding(0) var<storage, read> _A : array<${dataType}>;
+          @group(0) @binding(1) var<storage, read_write> output : array<i32>;
+
+          ${outputIndicesHelper.o2iImpl}
+          ${inputIndicesHelper.i2oImpl}
+
+          ${shaderHelper.mainStart()}
+          ${shaderHelper.guardAgainstOutOfBoundsWorkgroupSizes(outputSize)}
+          ${inputIndicesHelper.indicesVariableDeclaration('inputIndices')}
+          ${outputIndicesHelper.indicesVariableDeclaration('outputIndices')}
+          ${outputIndicesHelper.o2iCall('global_idx', 'outputIndices')}
+
+          ${idxCopy.join('\n')}
+          ${ops[0]}       // init ops
+          ${initInputIdx}
+          ${ops[1]}
+          ${reduceOps}
+          ${ops[3]} // final values
+          output[global_idx*2] = bestIndex; // result it int64
```

Comment:
Should this index be `global_idx*2 + 1` 
because the 32-bit result needs to go into the least significant 32-bits out of 64-bits?

Reply:
should be ok:
![image](https://github.com/microsoft/onnxruntime/assets/22941064/09cbd012-0eb4-4dee-9224-c990377fabda)

unit test is happy with the numerics too.